pkg/server/cloud/azure/network.go

```diff
@@ -336,6 +361,11 @@ func (az *AzureClient) RemoveRoute(destinationCIDR string) error {
 }
 
 func (az *AzureClient) AddRoute(destinationCIDR, instanceID string) error {
+	if destinationCIDR == "" && instanceID == "" {
```

Comment:
This check is an `||` in the aws code.  Curious if you meant `||`

Reply:
Might want to check the `||`s and `&&`s in front of all the `AddRoute` and `DeleteRoute` implementations.

Reply:
Thanks, I fixed them.